pac4j-jwt/src/main/java/org/pac4j/jwt/credentials/authenticator/JwtAuthenticator.java

```diff
@@ -61,15 +71,20 @@ public void validate(TokenCredentials credentials) {
 
         try {
             // Parse the token
-            JWT jwt = JWTParser.parse(token);
+            final JWT jwt = JWTParser.parse(token);
 
             if (jwt instanceof SignedJWT) {
                 signedJWT = (SignedJWT) jwt;
             } else if (jwt instanceof EncryptedJWT) {
-                JWEObject jweObject = (JWEObject) jwt;
+                final JWEObject jweObject = (JWEObject) jwt;
```

Comment:
Here I would test that the `encryptionSecret` and not fallback to the `signingSecret` as the old constructor should have set both.